### PR TITLE
Fix PS-4513 (Inconsistent SELECT...ORDER BY results with euckr char set / euckr_bin collation) (5.6)

### PR DESCRIPTION
**Pull request by percona-ysorokin**
https://jira.percona.com/browse/PS-4513

Added missing reset 'end_range' statement in 'ha_innobase::reset_template()'
in 'ha_innodb.cc' which is present in the fix for Bug #23481444
"OPTIMISER CALL ROW_SEARCH_MVCC() AND READ THE INDEX APPLIED BY UNCOMMITTED ROWS"
(commit mysql/mysql-server@15dbdac4)
in 8.0 but not in 5.6 and 5.7.

Added new 'main.bug91091' MTR test case which simulates the problem with
different results of 'SELECT' statements.

### CODE REVIEW

**Comment by percona-ysorokin** [comment]
Jenkins build link
https://jenkins.percona.com/view/PS%205.6/job/percona-server-5.6-param/2194/

File: mysql-test/include/bug91091.inc [hunk 1]
```diff
@@ -0,0 +1,159 @@
+DELETE FROM t WHERE c2 = '*****12889aaTHLIN_ACQS_DT_S8' AND c1 = 'Eid4i*******11372012******haHLIN_PBCR_*****_******00';
```
Comment (laurynas-biveinis) [inline mysql-test/include/bug91091.inc]: Is it possible to reduce the testcase?

File: mysql-test/t/bug91091.test [hunk 2]
```diff
@@ -0,0 +1,60 @@
+--source include/have_innodb.inc
```
Comment (laurynas-biveinis) [inline mysql-test/t/bug91091.test]: Please move to innodb suite

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: mysql-test/t/bug91091.test (hunk 2 above)
Comment (percona-ysorokin) [reply mysql-test/t/bug91091.test]: Initially I thought that this problem was collation-related and I put the test case to the `main` suite. The problem turned out to be in InnoDB code. Moving to the `innodb` test suite.

**Comment by percona-ysorokin** [comment]
MTR test case moved to the `innodb` test suite. Long `INSERT` statement reformatted.
Updated Jenkins build link
https://jenkins.percona.com/view/PS%205.6/job/percona-server-5.6-param/2195

**Review by laurynas-biveinis — APPROVED** [review]
Please also upstream and PR to 8.0 seed branch

**Pull request merged**